hnix-store-core/src/System/Nix/Nar.hs

```diff
@@ -252,7 +259,7 @@ localPackNar effs basePath = Nar <$> localPackFSO basePath
                               <*> narFileSize effs path'
                               <*> narReadFile effs path'
         (True , _) -> fmap (Directory . Map.fromList) $ do
-          fs <- narListDir effs path'
+          fs <- filter (pathFilter . (path' </>)) <$> narListDir effs path'
```

Comment:
`filter` uses the predicate (`FilePathFilter`) to *include* elements that match. The Haddock comment on 249 suggests that `localPathNar'` *excludes* matches.

Reply:
Yep, I was not sure myself while wirting it, hence this confusion. But we have to scrape this whole implementation as the only filter that we want to use is a nix function, and hence wrapped in a MonadNix monad. This implementation cannot be used at all from what I have found.